### PR TITLE
Initial Commit for Backend Comment

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- User comment UI and Backend
- Mark the comment with (Author) if the user owns the comment and the post
- Added Delete Bottom
- PostComment service
- Comment model
- Migration

**SNIPPETS**
TO pr

### CODE REVIEW

File: src/app/Http/Controllers/PostCommentController.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     * Add Comment
+     */
+    public function store(CommentRequest $request, Post $post): RedirectResponse
+    {
+        $comment = new Comment([
+            'content' => $request->input('content'),
+            'user_id' => auth()->id(),
+        ]);
+
+        $post->comments()->save($comment);
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostCommentController.php]: i think this should be in service

File: src/app/Http/Controllers/PostCommentController.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        return redirect()->back()->with('success', 'Comment added successfully');
+    }
+
+    /**
+     * Delete Comment
+     */
+    public function delete(Comment $comment)
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostCommentController.php]:
```suggestion
    public function delete(Comment $comment): RedirectResponse

File: src/app/Services/CommentService.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+class CommentService {
+    /**
+     * Delete CommentS
+     */
+    function deleteComment(Comment $comment): Void
+    {
+        $comment->delete();
+    }
+}
```
Comment (mikco-hipe) [inline src/app/Services/CommentService.php]: new line

**Comment by PeterDMadrid** [comment]
Changed the delete logic. 
From only the owner of the comment can delete 
To: Allow the authenticated user to delete a comment only if they own the comment and also own the post to which the comment belongs,

**Review by mikco-hipe — APPROVED** [review]
GFEN 🙇

**Pull request merged**